control-plane/connect-inject/controllers/endpointsv2/endpoints_controller.go

```diff
@@ -285,6 +340,102 @@ func getServicePorts(service corev1.Service) []*pbcatalog.ServicePort {
 	return ports
 }
 
+func getEffectiveTargetPort(targetPort intstr.IntOrString, prefixedPods selectorPodData, exactNamePods selectorPodData) string {
```

Comment:
This func attempts to encapsulate the complicated edge cases for picking a target port name from container ports. The happy path is the early return directly below when the K8s service target port is a string, which will be used without modification.